distributions/aws/test/e2e/fixtures/storage_efs_dependencies.py

```diff
@@ -0,0 +1,233 @@
+import time
+import pytest
+import subprocess
+import boto3
+
+from e2e.utils.config import metadata
+from e2e.utils.cognito_bootstrap.common import load_cfg, write_cfg
+from e2e.conftest import region
+from e2e.fixtures.cluster import cluster
+from e2e.utils.utils import rand_name
+from e2e.utils.config import configure_resource_fixture
+from e2e.fixtures.cluster import associate_iam_oidc_provider, create_iam_service_account
+from e2e.utils.utils import (
+    rand_name,
+    get_iam_client,
+    get_eks_client,
+    get_ec2_client,
+    get_efs_client,
+    curl_file_to_path,
+    get_aws_account_id,
+    kubectl_apply,
+    kubectl_delete,
+    kubectl_apply_kustomize,
+    kubectl_delete_kustomize,
+)
+
+DEFAULT_NAMESPACE = "kube-system"
+
+
+def wait_on_efs_status(desired_status, efs_client, file_system_id):
+    filesystem_status = ""
+    while filesystem_status != desired_status:
+        response = efs_client.describe_file_systems(
+            FileSystemId=file_system_id,
+        )
+        filesystem_status = response["FileSystems"][0]["LifeCycleState"]
+        print(f"{file_system_id} {filesystem_status} .... waiting")
+        time.sleep(10)
+
+
+@pytest.fixture(scope="class")
+def install_efs_csi_driver(metadata, region, request, cluster):
+    EFS_CSI_DRIVER = "github.com/kubernetes-sigs/aws-efs-csi-driver/deploy/kubernetes/overlays/stable/?ref=tags/v1.3.4"
+
+    def on_create():
+        kubectl_apply_kustomize(EFS_CSI_DRIVER)
+
+    def on_delete():
+        kubectl_delete_kustomize(EFS_CSI_DRIVER)
+
+
+@pytest.fixture(scope="class")
+def create_iam_policy(metadata, region, request, cluster):
+    # TODO: Existing IAM Client with Region does not seem to work.
+    efs_deps = {}
+    iam_client = boto3.client("iam")
+
+    EFS_IAM_POLICY = "https://raw.githubusercontent.com/kubernetes-sigs/aws-efs-csi-driver/v1.3.4/docs/iam-policy-example.json"
+    policy_name = rand_name("efs-iam-policy-")
+    aws_account_id = get_aws_account_id()
+    policy_arn = [f"arn:aws:iam::{aws_account_id}:policy/{policy_name}"]
+
+    def on_create():
+        associate_iam_oidc_provider(cluster, region)
+        curl_file_to_path(EFS_IAM_POLICY, "iam-policy-example.json")
+        with open("iam-policy-example.json", "r") as myfile:
+            policy = myfile.read()
+
+        response = iam_client.create_policy(
+            PolicyName=policy_name,
+            PolicyDocument=policy,
+        )
+
+        create_iam_service_account(
+            "efs-csi-controller-sa", DEFAULT_NAMESPACE, cluster, region, policy_arn
+        )
+        efs_deps["efs_iam_policy_name"] = policy_name
+
+    def on_delete():
+        details_efs_deps = metadata.get("efs_deps")
+        policy_name = details_efs_deps["efs_iam_policy_name"]
+        response = iam_client.delete_policy(
```

Comment:
same here

Reply:
Good catches, I basically was copy pasting commands and missed removing this. Done.